### DISCUSSION TITLE
租户的隔离策略

### DISCUSSION THREAD

**Discussion by GitHub-Stephen** [zuihou/zuihou-admin-cloud#263] [02.提问]
hello，zui大，想问问，我查看了源码中的gateway策略，看有灰度发布策略，但没了解租户的隔离策略是怎么实现的，可否指点迷津，感谢。

**Comment by zuihou** [comment]
- DATASOURCE模式：  一个租户一个独立数据库   
- SCHEMA 模式：一个租户一个数据库SCHEMA  
- COLUMN模式：通过租户ID隔离数据
- NONE模式：非租户模式

**Reply by zuihou** [reply]
实现方式都是通过注解  + 拦截器解析的方式来实现动态切换数据源或动态修改表表的条件
